Source/CBLDatabase+Insertion.m

```diff
@@ -228,8 +228,8 @@ - (CBLStatus) forceInsert: (CBL_Revision*)inRev
         NSString* prevRevID = history.count >= 2 ? history[1] : nil;
         if (![self processAttachmentsForRevision: updatedRev
                                        prevRevID: prevRevID
-                                          status: &status])
-        return status;
+                                          status: &status] && CBLStatusIsError(status))
```

Comment:
Oh, I see —it's because -mutateAttachments returns NO if there were no changes. So the right fix is to make -processAttachmentsForRevision ignore the return value of -mutateAttachments, and instead return `!CBLStatusIsError(*outStatus)`.